lisa/tests/scheduler/load_tracking.py

```diff
@@ -1087,51 +1088,60 @@ def get_rtapp_profile(cls, plat_info):
             profile[task] = RTATask()
 
-        for i in range(2):
+        common_phase_settings = dict(
+            duration_s=cls.PHASE_DURATION_S,
+            period_ms=cls.TASK_PERIOD_MS,
+        )
+
+        for cpu in cpus:
             # A task that will migrate to another CPU
             profile["migrating"] += Periodic(
-                duty_cycle_pct=cls.unscaled_utilization(plat_info, cpus[i], 20),
-                duration_s=cls.PHASE_DURATION_S, period_ms=cls.TASK_PERIOD_MS,
-                cpus=[cpus[i]])
+                duty_cycle_pct=cls.unscaled_utilization(plat_info, cpu, 20),
+                cpus=[cpu], **common_phase_settings)
 
             # Just some tasks that won't move to get some background utilization
             profile["static0"] += Periodic(
                 duty_cycle_pct=cls.unscaled_utilization(plat_info, cpus[0], 30),
-                duration_s=cls.PHASE_DURATION_S, period_ms=cls.TASK_PERIOD_MS,
-                cpus=[cpus[0]])
+                cpus=[cpus[0]], **common_phase_settings)
 
             profile["static1"] += Periodic(
                 duty_cycle_pct=cls.unscaled_utilization(plat_info, cpus[1], 20),
-                duration_s=cls.PHASE_DURATION_S, period_ms=cls.TASK_PERIOD_MS,
-                cpus=[cpus[1]])
+                cpus=[cpus[1]], **common_phase_settings)
 
         return profile
 
-class TwoTasksCPUMigration(OneTaskCPUMigration):
+class NTasksCPUMigrationBase(CPUMigrationBase):
     """
-    Two tasks on two big CPUs, swap their CPU in the second phase
+    N tasks on N CPUs, with all the migration permutations.
     """
 
     @classmethod
     def get_rtapp_profile(cls, plat_info):
-        profile = {}
         cpus = cls.get_migration_cpus(plat_info)
+        make_name = lambda i: 'migrating{}'.format(i)
 
-        for task in ["migrating0", "migrating1"]:
-            # An empty RTATask just to sum phases up
-            profile[task] = RTATask()
-
-        for i in range(2):
-            # A task that will migrate from CPU A to CPU B
-            profile["migrating0"] += Periodic(
-                duty_cycle_pct=20, duration_s=cls.PHASE_DURATION_S,
-                period_ms=cls.TASK_PERIOD_MS, cpus=[cpus[i]])
+        profile = {
+            make_name(i): RTATask()
+            for i, cpu in enumerate(cpus)
+        }
 
-            # A task that will migrate from CPU B to CPU A
-            profile["migrating1"] += Periodic(
-                duty_cycle_pct=20, duration_s=cls.PHASE_DURATION_S,
-                period_ms=cls.TASK_PERIOD_MS, cpus=[cpus[1 - i]])
+        # Define one task per CPU, and create all the possible migrations by
+        # shuffling around these tasks
+        for cpus_combi in itertools.permutations(cpus, r=len(cpus)):
+            for i, cpu in enumerate(cpus_combi):
+                profile[make_name(i)] += Periodic(
+                    duty_cycle_pct=20, duration_s=cls.PHASE_DURATION_S,
```

Comment:
As discussed offline, this should be changed scale on the number of requested tasks. Since we're still only using the 1 and 2 tasks definitions, that can be done in a later change.

Reply:
Done in the new version of the PR